src/modules/share/share.repository.ts

```diff
@@ -238,6 +243,57 @@ export class SequelizeShareRepository implements ShareRepository {
     await this.shareModel.destroy({ where: { id: shareId } });
   }
 
+  async findAllSharedFoldersToAUserPaginated(
+    user: User,
+    page: number,
+    perPage: number,
+    orderBy?: OrderBy,
+  ): Promise<Folder[]> {
+    const { offset, limit } = Pagination.calculatePagination(page, perPage);
+
+    const validTypePaginationParameters =
+      (isNaN(page) && !isNaN(perPage)) || (!isNaN(page) && isNaN(perPage));
+
+    if (validTypePaginationParameters) {
+      throw new BadRequestException('take and skip must be provided together');
+    }
+
+    const order = orderBy
+      ? [orderBy.split(':') as [string, string]]
+      : undefined;
+
+    const findCriteria = {
+      where: {
+        user_id: user.id,
+        is_folder: true,
+      },
+      include: [
+        {
+          model: this.folderModel,
+          where: {
+            deleted: false,
+          },
+          required: false,
+        },
+      ],
+    };
+
+    if (!isNaN(limit) && !isNaN(offset)) {
+      findCriteria['limit'] = limit;
+      findCriteria['offset'] = offset;
+    }
+
+    if (order) {
+      findCriteria['order'] = order;
+    }
+
+    const shares = await this.shareModel.findAll(findCriteria);
```

Comment:
According to the confluence documentation, we should return **the private folders** shared by a user with another user, shares table is for public sharing